src/AspNetCoreModuleV2/InProcessRequestHandler/inprocesshandler.cpp

```diff
@@ -59,6 +61,65 @@ IN_PROCESS_HANDLER::OnExecuteRequestHandler()
     return m_pApplication->OnExecuteRequest(m_pW3Context, this);
 }
 
+// static
+void * IN_PROCESS_HANDLER::operator new(size_t)
+{
+    DBG_ASSERT(sm_pAlloc != NULL);
+    if (sm_pAlloc == NULL)
+    {
+        return NULL;
+    }
+    return sm_pAlloc->Alloc();
+}
+
+// static
+void IN_PROCESS_HANDLER::operator delete(void * pMemory)
+{
+    DBG_ASSERT(sm_pAlloc != NULL);
+    if (sm_pAlloc != NULL)
+    {
+        sm_pAlloc->Free(pMemory);
+    }
+}
+
+// static
+HRESULT
+IN_PROCESS_HANDLER::StaticInitialize( VOID )
```

Comment:
I don't like the static initialize pattern; it's easy to forget calling it. Calling it in dll main also seems odd. Can we put this in the inprocess application?

Reply:
the sm_pAlloc is a static variable shared cross applications in the same worker process. We have to call some static initialize function to init it.